packages/vinxi/lib/plugins/tree-shake.js

```diff
@@ -1,6 +1,6 @@
-import jsx from "@babel/plugin-syntax-jsx";
```

Comment:
I worry that when we remove the direct import, these plugins are not found in different package managers. I dont know if babel manages that internally well. Can we do the direct import? Does it cause problems

Reply:
JSX itself is built into Babel. The problem is we don't need to leverage Babel to compile both JSX and TypeScript, we can delegate these to ESBuild.

Reply:
the syntax plugins are what you are adding in parserOpts.plugins too. The only difference if right now I import the syntax plugins and add them explicitly. Whereas the new method will make babel look for those syntax plugins.

Reply:
Actually the specific difference is in the transformation output. Adding the syntax plugins will compile/transform these into their vanilla JS code, not preserve them. Babel already knows about the syntaxes, they are just disabled like a feature flag.

This kind of implementation is not new btw, kind of derived this from Preact and implemented this on all of my babel-related libraries

Reply:
ahhh okay. thats great then!